### PR TITLE
Device automation

### PR DESCRIPTION
**Pull request by aik24**

### CODE REVIEW

File: cypress-E2E-tests/cypress/integration/device_property.cy.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    cy.apiCreateSimpleAsset(assetObject);
+    cy.addAssetToDevice('Device3', 'Asset Test');
+  });
+
+  beforeEach(function () {
+    cy.login();
+    cy.visitAndWaitUntilPageLoad(url);
+    cy.get(asset_properties_widget_elements.addWidgetButton).click();
+    cy.get(asset_properties_widget_elements.cardElement).eq(0).click();
+    cy.get(unassignedDevice).click();
```
Comment (dchc01) [inline cypress-E2E-tests/cypress/integration/device_property.cy.ts]: Here you can use clickOnAsset command.

File: cypress-E2E-tests/cypress/integration/device_property.cy.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  // User should be able to select a nested property for a device property "Network" and Verify that on view the subchildrens are visible.
+  it('TC_Device_Properties_004', () => {
+    cy.chooseAssetOrDevice('Device1');
+    cy.get(asset_properties_widget_elements.addPropertyButton).click();
+    cy.selectProperty('Network');
+    cy.get(asset_properties_widget_elements.selectButton).click();
+    cy.get(asset_properties_widget_elements.saveButton).click();
+    cy.contains('Network');
+    cy.contains('Address range');
+    cy.contains('DNS 2');
```
Comment (dchc01) [inline cypress-E2E-tests/cypress/integration/device_property.cy.ts]: See if you can get the specific element to add assertion.

**Review by dchc01 — CHANGES_REQUESTED** [review]

File: cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts [hunk 3]
```diff
@@ -572,4 +572,106 @@ describe('Device properties widget', function () {
...
+      cy.login();
+      cy.visitAndWaitUntilPageLoad(url);
+      cy.get(asset_properties_widget_elements.addWidgetButton).click();
+      cy.get(asset_properties_widget_elements.cardElement).eq(0).click();
+      cy.intercept({
+        method: 'GET',
+        url: '**/inventory/managedObjects?**'
+      }).as('loadDevices');
+      cy.get(unassignedDevice).click();
+      cy.wait('@loadDevices');
```
Comment (dchc01) [inline cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts]: Verify the status code.

**Review by dchc01 — CHANGES_REQUESTED** [review]

File: cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts (hunk 3 above)
Comment (aik24) [reply cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts]: Added

**Review by dchc01 — APPROVED** [review]

**Review by amar-c8y — APPROVED** [review]

**Pull request merged**